File: app/models/Notebook.java
package models;

import java.util.List;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.springframework.data.mongodb.core.index.Indexed;

import play.modules.spring.Spring;
import repositories.NoteRepository;


@JsonIgnoreProperties("notes")
public class Notebook extends AbstractAuditable{

	private String id;

	private String title;

	@Indexed
	private String userEmail;
	
	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public List<Note> getNotes() {
		return Spring.getBeanOfType(NoteRepository.class).findByNotebookId(id);
	}

	public String getUserEmail() {
		return userEmail;
	}

	public void setUserEmail(String userEmail) {
		this.userEmail = userEmail;
	}
}

File: server/note/src/main/kotlin/fr/acle/notello/note/model/Notebook.kt
package fr.acle.notello.note.model

import fr.acle.notello.common.model.BaseAuditableVersionableDocument
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Notebook(
        var title: String,
        @Indexed
        val userEmail: String
) : BaseAuditableVersionableDocument() {

    @Id
    lateinit var id: String

}